src/earn/prepareTransactions.ts

```diff
@@ -118,100 +123,169 @@ export async function prepareWithdrawAndClaimTransactions({
       ],
       ...shortcutTriggerArgs?.withdraw,
     })
+  Logger.debug(TAG, 'prepareWithdrawTransactions', {
+    withdrawTransactions,
+    pool,
+  })
+
+  // Is there a better way to do this?
+  // Pass a param to a shared prepareTransactions function to indicate that the transaction should also return swapTransaction:
+  return {
+    prepareTransactionsResult: await prepareTransactions({
+      feeCurrencies,
+      baseTransactions: rawShortcutTransactionsToTransactionRequests(withdrawTransactions),
+      isGasSubsidized: isGasSubsidizedForNetwork(networkId),
+      origin: 'earn-withdraw',
+    }),
+    swapTransaction: undefined,
```

Comment:
Does this `swapTransaction: undefined` actually need to be here?

Reply:
It is otherwise we get a type error on `usePrepareDepositAndWithdrawTransactions` in `EarnEnterAmount.tsx`: 
```
Property 'swapTransaction' does not exist on type '{ prepareTransactionsResult: PreparedTransactionsResult; } | {}'
```